packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/representations/DomainDropTreeItemHandler.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.studio.services.representations;
+
+import org.eclipse.sirius.components.collaborative.trees.api.IDropTreeItemHandler;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.trees.Tree;
+import org.eclipse.sirius.components.view.emf.IRepresentationDescriptionIdProvider;
+import org.eclipse.sirius.components.view.emf.IViewRepresentationDescriptionSearchService;
+import org.eclipse.sirius.web.application.views.explorer.services.ExplorerDescriptionProvider;
+import org.eclipse.sirius.web.application.views.explorer.services.ExplorerDropTreeItemHandler;
+import org.eclipse.sirius.web.domain.services.api.IMessageService;
+import org.springframework.stereotype.Service;
+
+/**
+ * Provides the drop tree item tool for the Domain explorer by DSL explorer.
+ *
+ * @author gdaniel
+ */
+@Service
+public class DomainDropTreeItemHandler extends ExplorerDropTreeItemHandler implements IDropTreeItemHandler {
+
+    private final IViewRepresentationDescriptionSearchService viewRepresentationDescriptionSearchService;
+
+    public DomainDropTreeItemHandler(IObjectService objectService, IMessageService messageService, IViewRepresentationDescriptionSearchService viewRepresentationDescriptionSearchService) {
+        super(objectService, messageService);
+        this.viewRepresentationDescriptionSearchService = viewRepresentationDescriptionSearchService;
```

Comment:
Is there a reason why it needs to be nullable?

Reply:
No, I fixed it